src/aggregate/aggregate_exec.c

```diff
@@ -619,6 +608,9 @@ static void sendChunk_Resp3(AREQ *req, RedisModule_Reply *reply, size_t limit,
 done_3:
     RedisModule_Reply_ArrayEnd(reply); // >results
 
+    // <total_results>
+    RedisModule_ReplyKV_LongLong(reply, "total_results", req->qiter.totalResults);
+
```

Comment:
isn't it going to change the reply fields' order?

Reply:
Mentioned in the PR comment, only for RESP3 where the order doesn't matter AFAIK

Reply:
🤔 i didn't know that

Reply:
anyway, is  `qiter.totalResults`  derived from the results array length to give the most accurate value?

Reply:
Kind of.. But more no than yes, unfortunately.
Take a look in the code, it's incremented and decremented on the fly as results get propagated or discarded in the pipeline flow.
It may be more accurate on the coordinator than on the shard level, since there it is taking the length of the `results` component as the base value.
This needs more attention in order to be 100% accurate, but I didn't want to break any existing behavior since this PR is meant to go to the version branches in a patch as well.

We will need to have a continuation to regularize this for all paths and pipelines, that will probably break some things..

Reply:
It will give a different value, that will be correct in more scenarios than the current state, relatively for free due to the map nature of the response. The "free" part is mainly since it is not breaking, and also low effort as you can see.

For instance, today using the strict timeout policy we will get a correct value in both implementations with high probability, but using the non-strict policy we almost never get a correct value in the current state. After the fix, we will get a correct value for that case instead of the non-informative `1` value that we got until now.

Reply:
@raz-mon Nitpick - if `total_results` is still not guaranteed to always be accurate, although being better than the current value, will users be able to rely on it (with RESP3)?

Reply:
More than they could yesterday..
As mentioned in [this comment](https://github.com/RediSearch/RediSearch/pull/5765#discussion_r1997637337), we will need to do a more complete and probably breaking work to make sure this value is not broken in some paths